src/nnet3/nnet-simple-component.cc

```diff
@@ -1136,8 +1136,13 @@ void RectifiedLinearComponent::StoreStats(
 }
 
 void AffineComponent::Scale(BaseFloat scale) {
-  linear_params_.Scale(scale);
-  bias_params_.Scale(scale);
+  if (scale == 0.0) {
```

Comment:
you could comment:
// If scale == 0.0 we call SetZero() which will get rid of NaN's and inf's.